src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts

```diff
@@ -79,6 +79,26 @@ export class RefreshAction extends Action {
 	}
 }
 
+export class EditConnectionAction extends Action {
+	public static ID = 'registeredServers.editConnection';
+	public static LABEL = localize('connectionTree.editConnection', "Edit Connection");
+
+	constructor(
+		id: string,
+		label: string,
+		private _connectionProfile: ConnectionProfile,
+		@IConnectionManagementService private _connectionManagementService: IConnectionManagementService
+	) {
+		super(id, label);
+		this.class = 'edit-server-action';
+	}
+
+	public async run(element: ConnectionProfileGroup): Promise<boolean> {
+		await this._connectionManagementService.showConnectionDialog(undefined, undefined, this._connectionProfile);
```

Comment:
do you think this change will be sufficient to address the request?  For example, if I change things like the server or database won't I get two connection nodes in OE?

Reply:
I agree. People can already essentially do this functionality by selecting the connection from the "Recently Used" list when creating a new connection (and then modifying the fields as appropriate) so while adding an action to do it helps make that a bit easier that doesn't address the main ask of the enhancement requests.

Reply:
It works for changing the optional name of the server connection, but I will need to test those features out

Reply:
Do I need to write a new connection widget to handle editing the server details? I don't think you can use the existing one

Reply:
Why not? Sure the current widget doesn't currently support actually editing an existing connection in-place but the actual design of the dialog shouldn't need to change. You could just add a mode (edit/add/etc) that determines whether we copy and create a new connection or modify the connection passed in.

Reply:
@smartguest I think connections should have a unique GUID to identify them.  So when the Connection Dialog edits an existing connection, that connection profile metadata should be updated.  So if the user changes the server or database name, the existing connection should be changed to reflect those values in Object Explorer, etc.  

Other UI, such as Query editors, notebooks, or dashboards that are using that are actively connection should not be impacted.  Only new connections created from that profile should use the updated values.  Please sync-up with me if the intended use-case isn't clear here.